src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestRunConfigurationProducer.kt

```diff
@@ -23,83 +24,141 @@ class CargoTestRunConfigurationProducer : RunConfigurationProducer<CargoCommandC
-
         configuration.name = test.configurationName
-        val cmd = test.cargoCommandLine.mergeWithDefault(configuration)
+        val commandLine = test.cargoCommandLine() ?: return false
+        val cmd = commandLine.mergeWithDefault(configuration)
         configuration.setFromCmd(cmd)
         return true
     }
 
-
     companion object {
-        fun findTest(psi: PsiElement, climbUp: Boolean = true): TestConfig? =
-            findTestFunction(psi, climbUp) ?: findTestMod(psi, climbUp)
-
-        private fun findTestFunction(psi: PsiElement, climbUp: Boolean): TestConfig? {
-            val fn = findElement<RsFunction>(psi, climbUp) ?: return null
-            val name = fn.crateRelativePath.configPath() ?: return null
-            val target = fn.containingCargoTarget ?: return null
-            return if (fn.isTest) TestConfig(fn, "Test $name", name, target, true) else null
+        fun findTest(context: ConfigurationContext): TestConfig? {
+            val elements: Array<PsiElement> = LangDataKeys.PSI_ELEMENT_ARRAY.getData(context.dataContext)
+                ?: context.location?.psiElement?.let { arrayOf(it) } ?: return null
+            return findTest(elements)
         }
 
-        private fun findTestMod(psi: PsiElement, climbUp: Boolean): TestConfig? {
-            val mod = findElement<RsMod>(psi, climbUp) ?: return null
-
-            val testName = if (mod.modName == "test" || mod.modName == "tests")
-                "Test ${mod.`super`?.modName}::${mod.modName}"
-            else
-                "Test ${mod.modName}"
-            val testPath = mod.crateRelativePath.configPath() ?: ""
-            val target = mod.containingCargoTarget ?: return null
-            if (!hasTestFunction(mod)) return null
-
-            return TestConfig(mod, testName, testPath, target, false)
-        }
+        fun findTest(psi: Array<PsiElement>, climbUp: Boolean = true): TestConfig? =
+            when (psi.size) {
+                0 -> null
+                else ->
+                    TestConfig.MultipleFileTestConfig.create(psi.mapNotNull {
+                        findElement<RsMod>(it, climbUp)
+                    }.toTypedArray()) ?:
+
+                    findElement<RsFunction>(psi[0], climbUp)?.let {
+                        TestConfig.SingleTestConfig.create(it)
+                    } ?:
+
+                    findElement<RsMod>(psi[0], climbUp)?.let {
+                        TestConfig.SingleTestConfig.create(it)
+                    }
+            }
 
         private inline fun <reified T : PsiElement> findElement(base: PsiElement, climbUp: Boolean): T? {
             if (base is T) return base
             if (!climbUp) return null
             return base.ancestorOrSelf()
         }
+    }
+}
+
+
+
+sealed class TestConfig {
+    companion object {
+        private fun hasTestFunction(mod: RsMod): Boolean =
+            mod.processExpandedItems { it is RsFunction && it.isTest }
```

Comment:
In this case:
```rust
// tests/foo.rs
mod bar {
    #[test]
    fn baz(){}
}
```
we won't run tests for `foo` mod.
But I think it is not very important because looks like it's uncommon code style for tests